components/common/src/main/java/com/hotels/styx/common/content/FlowControllingHttpContentProducer.java

```diff
@@ -70,6 +83,9 @@
 
     private volatile Subscriber<? super ByteBuf> contentSubscriber;
 
+    private final HashedWheelTimer timer = new HashedWheelTimer();
```

Comment:
From HashedWheelTimer: 

```
Do not create many instances.
HashedWheelTimer creates a new thread whenever it is instantiated and started. Therefore, you should make sure to create only one instance and share it across your application. One of the common mistakes, that makes your application unresponsive, is to create a new instance for every connection.

```
However, it seems we are creating a Timer for every http request we receive.